sys/include/checksum/crc16_ccitt.h

```diff
@@ -39,6 +36,8 @@ extern "C" {
 
 /**
  * @brief           Update CRC16-CCITT
+ * @deprecated      Use @ref crc16_ccitt_false_update instead. 
+ * Will be removed after 2023.01 release.
```

Comment:
> This might lead to confusion for users that use the old ccitt. Maybe, it is best to move this through proper API aliasing and deprecation, i.e. introduce new crc16_ccitt_*() inline functions that wrap around crc16_ccitt_aug_* and add a @deprecation note that this function will be removed in 2 release cycles (i.e. 2023.01 if this PR gets merged soon). That node could also include a mention that crc16_ccitt_aug_* should be used instead.

@miri64 like this? :)